File: win-prog/ProgWinVB/Hello Windows Forms/InstantiateHelloWorld/InstantiateHelloWorld.vb
'------------------------------------------------------
' InstantiateHelloWorld.vb (c) 2002 by Charles Petzold
'------------------------------------------------------
Imports System
Imports System.Drawing
Imports System.Windows.Forms

Module InstantiateHelloWorld
    Sub Main()
        Dim frm As New HelloWorld()
        frm.Text = "Instantiate " & frm.Text
        AddHandler frm.Paint, AddressOf MyPaintHandler
        Application.Run(frm)
    End Sub

    Sub MyPaintHandler(ByVal obj As Object, ByVal pea As PaintEventArgs)
        Dim frm As Form = DirectCast(obj, Form)
        Dim grfx As Graphics = pea.Graphics
        grfx.DrawString("Hello from InstantiateHelloWorld!", _
                        frm.Font, Brushes.Black, 0, 100)
    End Sub
End Module

File: win-prog/ProgWinCS/Hello Windows Forms/InstantiateHelloWorld/InstantiateHelloWorld.cs
//----------------------------------------------------
// InstantiateHelloWorld.cs © 2001 by Charles Petzold
//----------------------------------------------------
using System;
using System.Drawing;
using System.Windows.Forms;

class InstantiateHelloWorld
{
     public static void Main()
     {
          Form form   = new HelloWorld();
          form.Text   = "Instantiate " + form.Text;
          form.Paint += new PaintEventHandler(MyPaintHandler);

          Application.Run(form);
     }
     static void MyPaintHandler(object objSender, PaintEventArgs pea)
     {
          Form     form = (Form)objSender;
          Graphics grfx = pea.Graphics;

          grfx.DrawString("Hello from InstantiateHelloWorld!", 
                          form.Font, Brushes.Black, 0, 100);
     }
}